lib/puppet/provider/mysql.rb

```diff
@@ -107,6 +107,8 @@ def self.cmd_options(options)
     options.each do |opt|
       if opt == 'GRANT'
         option_string << ' WITH GRANT OPTION'
+      elsif op = opt.match(/^REQUIRE\s(SSL|X509)$/)
```

Comment:
well, we already know that if SSL or X509 is given, it's a REQUIRE. Why not take a similar approach than 'GRANT' and just use 'SSL' or 'X509'; Those will be looked for here and subsequently the option_string will fill in the REQUIRE. So something like:
```
elsif opt == 'SSL' or opt == 'X509'
  option_string << "REQUIRE #{opt}"
```